test/hpke_test.c

```diff
@@ -0,0 +1,1728 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+#include <openssl/core_names.h>
+#include <openssl/rand.h>
+#include <openssl/hpke.h>
+#include "testutil.h"
+
+/* a size to use for stack buffers */
+#define OSSL_HPKE_TSTSIZE 512
+
+static OSSL_LIB_CTX *testctx = NULL;
+static char *testpropq = NULL;
+
+/**
+ * @brief Test that an EVP_PKEY encoded public key matches the supplied buffer
+ * @param pkey is the EVP_PKEY we want to check
+ * @param pub is the expected public key buffer
+ * @param publen is the length of the above
+ * @return 1 for good, 0 for bad
+ */
+static int cmpkey(const EVP_PKEY *pkey,
```

Comment:
shift this function down after the typedefs

Reply:
done